store/v2alpha1/smt/store.go

```diff
@@ -77,6 +77,17 @@ func (s *Store) GetProofICS23(key []byte) (*ics23.CommitmentProof, error) {
 	return createIcs23Proof(s, key)
 }
 
+func (s *Store) GetSMTProof(key []byte) (*smt.SparseMerkleProof, error) {
+	if len(key) == 0 {
+		return nil, errKeyEmpty
+	}
+	proof, err := s.tree.Prove(key)
+	if err != nil {
+		return nil, err
+	}
+	return &proof, nil
+}
```

Comment:
What is wrong with above's `GetProof`?

Reply:
The smt library accepts a raw proof only in the construction of deepSubTrees which are used when verifying a fraudProof before loading up a baseApp with it

Reply:
Hmm but this is about _reading_ proofs, not _writing_ proofs, no? Why does the SMT accepting raw proofs as writes matter?